smartsim/_core/mli/workermanager.py

```diff
@@ -0,0 +1,976 @@
+
+        for k, v in zip(request.output_keys, execute_result.predictions):
+            feature_store[k] = v
+            keys.append(k)
+
+        return keys
+
+
+class MachineLearningWorkerBase(MachineLearningWorkerCore, ABC):
+    """Abstrct base class providing contract for a machine learning
+    worker implementation."""
+
+    @staticmethod
+    @abstractmethod
+    def deserialize(data_blob: bytes) -> InferenceRequest:
+        """Given a collection of data serialized to bytes, convert the bytes
+        to a proper representation used by the ML backend
+        :param data_blob: inference request as a byte-serialized blob
+        :return: InferenceRequest deserialized from the input"""
+
+    @staticmethod
+    @abstractmethod
+    def load_model(
+        request: InferenceRequest, fetch_result: FetchModelResult
+    ) -> ModelLoadResult:
+        """Given a loaded MachineLearningModel, ensure it is loaded into
+        device memory
+        :param request: The request that triggered the pipeline
+        :return: ModelLoadResult wrapping the model loaded for the request"""
+
+    @staticmethod
+    @abstractmethod
+    def transform_input(
+        request: InferenceRequest, fetch_result: InputFetchResult
+    ) -> InputTransformResult:
+        """Given a collection of data, perform a transformation on the data
+        :param request: The request that triggered the pipeline
+        :param fetch_result: Raw output from fetching inputs out of a feature store
+        :return: The transformed inputs wrapped in a InputTransformResult"""
+
+    @staticmethod
+    @abstractmethod
+    def execute(
+        request: InferenceRequest,
+        load_result: ModelLoadResult,
+        transform_result: InputTransformResult,
+    ) -> ExecuteResult:
+        """Execute an ML model on inputs transformed for use by the model
+        :param request: The request that triggered the pipeline
+        :param load_result: The result of loading the model onto device memory
+        :param transform_result: The result of transforming inputs for model consumption
+        :return: The result of inference wrapped in an ExecuteResult"""
+
+    @staticmethod
+    @abstractmethod
+    def transform_output(
+        request: InferenceRequest,
+        execute_result: ExecuteResult,
+    ) -> OutputTransformResult:
+        """Given inference results, perform transformations required to
+        transmit results to the requestor.
+        :param request: The request that triggered the pipeline
+        :param execute_result: The result of inference wrapped in an ExecuteResult
+        :return:"""
+
+    @staticmethod
+    @abstractmethod
+    def serialize_reply(reply: InferenceReply) -> bytes:
+        """Given an output, serialize to bytes for transport
+        :param reply: The result of the inference pipeline
+        :return: a byte-serialized version of the reply"""
+
+
+class SampleTorchWorker(MachineLearningWorkerBase):
```

Comment:
Should we move this class to somewhere in the test directory?

Reply:
Previously I had mentioned maybe putting these into a file like ``tests/mli/mocks.py``, but now I'm wondering after seeing these objects if we would end up with an enormous python file.    Would it be better to have something like ``tests/mli/sample_feature_store.py`` and ``tests/mli/sample_worker``.  Open to suggestions but it might be nice to organize these a bit by file.

Reply:
- killed off SampleTorchWorker (only `IntegratedTorchWorker` was valuable) 
- moved all mock/test implementations into `tests/mli/*.py`